src/type_expr.rs

```diff
@@ -13,6 +19,20 @@ pub enum TypeInfo {
     Defined(DefinedTypeInfo),
 }
 
+impl TypeInfo {
+    /// Emit the type name with generics as an expression.
+    pub fn emit_expr<'ctx>(
+        &'static self,
+        w: &'ctx mut dyn io::Write,
+        options: &'ctx DefinitionFileOptions<'ctx>,
+    ) -> io::Result<()> {
```

Comment:
Neither of these arguments need to be behind a reference. Please follow the signature of `write_definition_file`:
```suggestion
    pub fn emit_expr<W>(
        &'static self,
        mut writer: W,
        options: DefinitionFileOptions<'_>,
    ) -> io::Result<()> {
```

Reply:
Good point, thanks